x-pack/platform/plugins/shared/entity_manager/server/lib/v2/definitions/install_built_in_definitions.ts

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { IClusterClient, Logger } from '@kbn/core/server';
+import { storeTypeDefinition } from './type_definition';
+import { storeSourceDefinition } from './source_definition';
+import { builtInDefinitions } from './built_in';
+
+export async function installBuiltInDefinitions(clusterClient: IClusterClient, logger: Logger) {
+  logger.info('Installing built in entity definitions');
+
+  for (const definition of builtInDefinitions) {
+    const { type, sources } = definition;
+
+    try {
+      await storeTypeDefinition({
+        type,
+        clusterClient,
+        logger,
+        replace: true,
+      });
+    } catch (error) {
+      logger.error(error.message);
```

Comment:
should this block be atomic ? is a source without its corresponding type a working or corrupt state ?

Reply:
It's not clear from this code but that logic is enforced inside the lib, if there is no matching type an error is thrown and the source won't be installed.

Now, the other way around is harder. Is a type without **any** source a valid state? I think not but I don't see how we can enforce that?

Reply:
> It's not clear from this code but that logic is enforced inside the lib, if there is no matching type an error is thrown and the source won't be installed.

can we add a comment to mention that this is enforced later on ?
so the point of not throwing here is that the installation could fail, but we may still have the type installed from a previous install so we can carry on with the sources installation ?

> Is a type without any source a valid state? I think not but I don't see how we can enforce that?

I think type with empty sources is fine, at least the v2 code already expects the situation

Reply:
> can we add a comment to mention that this is enforced later on ?

Yes, sir!

> so the point of not throwing here is that the installation could fail, but we may still have the type installed from a previous install so we can carry on with the sources installation ?

Exactly, since there might be more sources for that type in the pipeline.

> I think type with empty sources is fine, at least the v2 code already expects the situation

Great!